### PR TITLE
Unwrap metric_data

### LINKED ISSUE #5: Dimensions are not being correctly prefixed.
Metric data is all prefixed with `MetricData.member` correctly but dimensions are getting jammed together instead of being prefixed appropriately. 

This is the request params that are output from `put_metric_data/2`
```
%{
   "Action" => "PutMetricData",
   "MetricData.member.Dimensions.JobsCount" => "Jobs",
   "MetricData.member.MetricName" => "JobsCount",
   "MetricData.member.Unit" => "Count",
   "MetricData.member.Value" => 0,
   "Namespace" => "Dev/Jobs",
   "Version" => "2010-08-01"
 }
```

These are the expected params from [the docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html)
```
https://monitoring.&api-domain;/doc/2010-08-01/
?Action=PutMetricData
&Version=2010-08-01
&Namespace=TestNamespace
&MetricData.member.1.MetricName=buffers
&MetricData.member.1.Unit=Bytes
&MetricData.member.1.Value=231434333
&MetricData.member.1.Dimensions.member.1.Name=InstanceType
&MetricData.member.1.Dimensions.member.1.Value=m1.small
&AUTHPARAMS
```

So the dimensions need to be split between name and value and need their own prefix of `Dimensions.member.Name` and `Dimensions.member.Value`

### PR DESCRIPTION
**Pull request by CaldwellYSR**
`format_param/1` was not working correctly for nested fields. Each field
was being prefixed with MetricData.member but the inner fields were not
getting their specific prefixes. This caused dimensions to not be sent
correctly.

This fixes #5

### CODE REVIEW

File: lib/ex_aws/cloudwatch.ex
Comment (yumeto) [inline lib/ex_aws/cloudwatch.ex]: It seems not correct implementation to me...

Type spec of `metric_data` is `[metric_datum, ...]`, therefor the value passed to mapping function (`&format_param/1`) is like `[dimensions: [my_dim_1: "MyDimValue1", my_dim_2: "MyDimValue2"], metric_name: "MyTestMetic", timestamp: #DateTime<2018-06-20 02:51:43Z>, unit: "Count", value: 2]`

`Enum.flat_map(&format_param/1)` causes the error like this:

```
     ** (FunctionClauseError) no function clause matching in ExAws.Cloudwatch.format_param/1

     The following arguments were given to ExAws.Cloudwatch.format_param/1:
     
         # 1
         [dimensions: [my_dim_1: "MyDimValue1", my_dim_2: "MyDimValue2"], metric_name: "MyTestMetic", timestamp: #DateTime<2018-06-20 02:51:43Z>, unit: "Count", value: 2]
     
     Attempted function clauses (showing 10 out of 12):
     
         defp format_param({:alarm_actions, alarm_actions})
         defp format_param({:alarm_names, alarm_names})
         defp format_param({:dashboard_names, dashboard_names})
         defp format_param({:dimensions, dimensions})
         defp format_param({:end_time, end_time})
         defp format_param({:extended_statistics, extended_statistics})
         defp format_param({:insufficient_data_actions, insufficient_data_actions})
         defp format_param({:metric_data, metric_data})
         defp format_param({:ok_actions, ok_actions})
         defp format_param({:start_time, start_time})
```

Comment (CaldwellYSR) [reply lib/ex_aws/cloudwatch.ex]: @yumeto 

How did you get to that error? The below is my implementation that is working with the PR I posted above. 

```
    [
      dimensions: [{:jobs_count, "Jobs"}],
      metric_name: "JobsCount",
      unit: "Count",
      value: PgJobRunner.count_jobs()
    ]
    |> ExAws.Cloudwatch.put_metric_data(namespace)
    |> ExAws.request()
```

This is posting up to CloudWatch without any errors from my end using the branch referenced in the PR.

Comment (yumeto) [reply lib/ex_aws/cloudwatch.ex]: @CaldwellYSR I've passed the list of metric data

```
  [
     [
      dimensions: [{:jobs_count, "Jobs"}],
      metric_name: "JobsCount",
      unit: "Count",
      value: PgJobRunner.count_jobs()
      ],
     [
      dimensions: [{:jobs_count, "Jobs"}],
      metric_name: "OtherMetric",
      unit: "Count",
      value: 1
      ]
    ]
    |> ExAws.Cloudwatch.put_metric_data(namespace)
    |> ExAws.request()
```

Comment (CaldwellYSR) [reply lib/ex_aws/cloudwatch.ex]: Oh true I suppose I am passing only the one metric datum instead of a list. I will go back and reassess. Thanks

**Comment by CaldwellYSR** [comment]
@cammellos @yumeto and I fixed this so it should work correctly now.

**Comment by yumeto** [comment]
@cammellos I'll create a new PR for #7 after this PR was merged.

**Review by yumeto — APPROVED** [review]
I've reviewed the part written by @CaldwellYSR and he's done for the part I wrote as well.

@benwilson512 do you have a time to review and merge this PR?

**Comment by benwilson512** [comment]
@cammellos should be setup to review and merge, part of the point of splitting out each of these libraries is that I could let folks who are more closely familiar with these libraries handle those questions.

**Comment by cammellos** [comment]
@benwilson512 agreed, I think I might need write access to the repo though, I was not able to merge PRs before, just review them.
Thanks!

**Review by cammellos — APPROVED** [review]

**Comment by benwilson512** [comment]
Weird! I definitely tried to give you write access haha let me try again.

**Comment by benwilson512** [comment]
Ah, I think I messed up the master branch protection. Can you try now?

**Comment by CaldwellYSR** [comment]
@benwilson512 Might be worth checking the other ex_aws libraries for permissions issues when you have time. Just in case more of them have this problem.

**Comment by cammellos** [comment]
@benwilson512 works now, thank you!

**Comment by cammellos** [comment]
@benwilson512 one last thing, sorry to bother you again. 
I would like to set CI tests to run. I would not mind using circleci, you should see a request to authorize the app, let me know if that's ok or we should use a different CI system (travis?).
Thanks!

**Pull request merged**
